File: InterviewBit/Arrays/First Missing Integer.cpp
int Solution::firstMissingPositive(vector<int> &A) {
    int missing = 1;
    sort(A.begin(), A.end());
    
    for(auto num : A) {
        if(num < 1) { continue; }
            
        if(num == missing) {
            missing += 1;
            continue;
        }
        
        break;
    }
    
    return missing;
}

File: InterviewBit/Arrays/First Missing Integer.rb
class Solution
    # @param a : array of integers
    # @return an integer
    def firstMissingPositive(a)
        missing = 1
        a.sort!
        a.each { |num|
            next if num < 1
            missing += 1 and next if num == missing
            break
        }
        missing
    end
end
